Jenkinsfile

```diff
@@ -77,8 +77,11 @@ pipeline {
         stash allowEmpty: true, name: 'source', useDefaultExcludes: false
         script {
           dir("${BASE_DIR}"){
+            // Compare with the previous git commit if possible, otherwise the target branch or the git base commit.
+            def from = env.GIT_PREVIOUS_COMMIT?.trim() ? "${env.GIT_PREVIOUS_COMMIT}"
```

Comment:
@SylvainJuge brought up a good point in the Java agent weekly today: what if a docs update commit cancels an ongoing build? Would we then still skip the tests, even if the cancelled build has non-docs changes?

Reply:
That's a good question. 
The current implementation for `cancelOnGoingBuilds` and `gitRegionMatch` are not overlapped hence it could be the case.

Hence, we might need to revisit the `cancelOnGoingBuilds` with `gitRegionMatch` to evaluate if only-docs, then rerun everything too.  Only in case we wanna go smarter, otherwise, the existing implementation (the one in `main`) will not cause any problems.

Reply:
I've just pushed https://github.com/elastic/apm-agent-java/pull/2778/commits/b5a8ed52af07b7ea1e12ac5b574dc4dddc15c284

Builds on PRs for only-doc changes will never cancel any of the old existing builds in the CI, this should avoid the corner case mentioned above.

Reply:
```suggestion
            def from = env.GIT_PREVIOUS_COMMIT?.trim() ? "HEAD~1"
```